django/middleware/csrf.py

```diff
@@ -221,7 +222,30 @@ def process_view(self, request, callback, callback_args, callback_kwargs):
                 # branches that call reject().
                 return self._accept(request)
 
-            if request.is_secure():
+            # Verify the Origin header, if present.
+            origin_verified = False
+            if 'HTTP_ORIGIN' in request.META:
+                good_origin = '%s://%s' % (
+                    'https' if request.is_secure() else 'http',
+                    request.get_host(),
+                )
+                allowed_origins = [good_origin]
+                allowed_origins_regexes = []
+                for trusted_origin in settings.CSRF_TRUSTED_ORIGINS:
+                    if '*' in trusted_origin:
+                        trusted_origin_regex = '^' + trusted_origin.replace('*', r'[\w\-]+') + '$'
```

Comment:
This regex isn't necessarily complete. It may be better to do `urlparse(trusted_origin)` and check if the scheme matches and if `is_same_domain()` returns true, as referer checking does.